### ISSUE TITLE
filter_roidb(roidb) not found in train.py

### ISSUE THREAD

**Issue by saraghaie** [VisionLearningGroup/R-C3D#19]
Hello,

thank you for sharing your code.
The function filter_roidb() is not available or imported in the lib/tdcnn/train.py but called at line 117 of train.py
could you please provide this function or shall I just ignore it by commenting this line?

**Comment by huijuan88** [comment]
comment this line.


On Apr 5, 2018, at 11:03, saraghaie <notifications@github.com<mailto:notifications@github.com>> wrote:


Hello,

thank you for sharing your code.
The function filter_roidb() is not available or imported in the lib/tdcnn/train.py but called at line 117 of train.py
could you please provide this function or shall I just ignore it by commenting this line?

**Comment by saraghaie** [comment]
Thank you for your quick answer!

**Comment by kstys** [comment]
hello,I also come across this question,have you solve it? @saraghaie

**Comment by saraghaie** [comment]
@kstys yes, you can just comment the line and it will work :)

**Comment by kstys** [comment]
thanks a lot @saraghaie

**Comment by kstys** [comment]
and then a come across another question,

> import google.protobuf as pb2
>  pb2.text_format.Merge(f.read(), self.solver_param)

`AttributeError: 'module' object has no attribute 'text_format'`

and I look for some solution,but fail,have you some    suggestion??
 @saraghaie

**Comment by kstys** [comment]
thank you , @saraghaie 
and I have solve it, i add

> import google.protobuf.text_format
before

> import google.protobuf as pb2
it will work

**Comment by saraghaie** [comment]
Hi @kstys,
I didn't have any problem with import google.protobuf.text_format. Its perfect that you found the solution, it seems that some people also had the same issue after trying to use faster RCNN:
https://github.com/rbgirshick/py-faster-rcnn/issues/198

**Comment by kstys** [comment]
thanks a lot,haveyou train the it on THUMOS14?? @saraghaie

**Comment by saraghaie** [comment]
@kstys yes I trained on THUMOS14 and I could get approximately the same results as the authors. But I needed to change some parts of the code for myself, because it is at some parts hard coded. Feel free to let me know if there is anything that I can help with.

**Comment by jxiangli** [comment]
Hi, @saraghaie 
The results are lower than the authors using the model trained on THUMOS14 by myself. Did you change the training parameters? And could you please tell me what parts of code you changed in detail?

**Comment by kstys** [comment]
That's good ,and I want to train it on THUMOS14,but I can not find the file Class_Index_Detection.txt ,could you give me some hint and let me start the work ? @saraghaie

**Comment by saraghaie** [comment]
@kstys 
this file is provided with the THUMOS14 dataset, following is the content of Class_Index_Detection.txt

7 BaseballPitch
9 BasketballDunk
12 Billiards
21 CleanAndJerk
22 CliffDiving
23 CricketBowling
24 CricketShot
26 Diving
31 FrisbeeCatch
33 GolfSwing
36 HammerThrow
40 HighJump
45 JavelinThrow
51 LongJump
68 PoleVault
79 Shotput
85 SoccerPenalty
92 TennisSwing
93 ThrowDiscus
97 VolleyballSpiking

**Comment by saraghaie** [comment]
@jxiangli 
following are my settings in the td_cnn_end2end.yml. The changes that I applied were just for making the code run on my system, I don't know if such settings are necessary for everyone.

TRAIN:
  HAS_RPN: True
  VIDEO_BATCH: 1
  TWIN_NORMALIZE_TARGETS_PRECOMPUTED: True
  RPN_POSITIVE_OVERLAP: 0.7
  RPN_BATCHSIZE: 64
  PROPOSAL_METHOD: gt
  BG_THRESH_LO: 0.0
  RPN_MIN_SIZE: 8
  LENGTH: [512]
TEST:
  HAS_RPN: True
  NMS: 0.3
  RPN_MIN_SIZE: 8
  LENGTH: [512]

INPUT: "video"
NUM_CLASSES: 21

**Comment by kstys** [comment]
Thanks, I have sovle the all bug,and network has run,next,I will try to change some code @saraghaie

**Comment by jxiangli** [comment]
Thanks a lot!

**Comment by kstys** [comment]
Hi, @saraghaie 
I have a problem about dataset that I can not understand. when I use the dataset THUMOS14 and then split into frames. and in the test data ,it only includes 213 files,but I have 1574 videos ,and then I look for TH14_Temporal_Annotations_Test ,and I find that it maybe only include 213 annotation videos ,but what 's the meaning of  the remaining of 1361 videos ?

**Comment by ranranxu95** [comment]
how can i get this code?

**Comment by huijuan88** [comment]
The remaining videos may be used for other classification tasks.

It is in the preprocessing code.




On Jul 3, 2018, at 07:14, ranran <notifications@github.com<mailto:notifications@github.com>> wrote:


how can i get this code?
